### PR TITLE
Add functions to run notebooks automatically

### PR DESCRIPTION
**Pull request by antoine-daigneault-demers**

### CODE REVIEW

File: src/notebook_utils/automation.py [hunk 1]
```diff
@@ -14,6 +14,67 @@
...
 
+def run_notebook(notebook_path: PathLike):
+    return subprocess.run(["jupyter", "nbconvert", "--to", "notebook", "--inplace", "--execute", "--ExecutePreprocessor.timeout=600", f"{notebook_path}"], capture_output=True, text=True)
+
+def run_notebook_cli():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-f",
+        "--file",
+        type=pathlib.Path,
```
Comment (fmigneault) [inline src/notebook_utils/automation.py:44]: Is there a `File` type? I vaguely remember seeing that in docs to be more specific about the type of path allowed.

File: src/notebook_utils/automation.py [hunk 2]
```diff
@@ -14,6 +14,67 @@
...
+
+    parser.add_argument("-v", "--verbose", action="store_true", help="Output logging details.")
+    args = parser.parse_args()
+
+    if args.verbose:
+        _logger.setLevel(logging.INFO)
+
+    if args.file is None or args.file.suffix != ".ipynb":
+        _logger.error("You must specify a valid notebook (.ipynb file) to run.")
+        sys.exit(1)
```
Comment (fmigneault) [inline src/notebook_utils/automation.py]: You can define an action (inherit from `argparse.Action`, and pass it to with `action=Class` to `add_argument`) to perform all these kind of checks during `__call__`. If invalid, raise an argument error directly. 
See example: https://github.com/crim-ca/weaver/blob/e39ffccf3643a70b5496ca8b8799fa1a903c1a66/weaver/cli.py#L1490-L1499

File: src/notebook_utils/automation.py [hunk 3]
```diff
@@ -14,6 +14,67 @@
...
+
+    sys.exit(0)
+
+def run_notebooks_cli():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-d",
+        "--directory",
+        type=pathlib.Path,
```
Comment (fmigneault) [inline src/notebook_utils/automation.py]: Same as other, but possible `Directory` type?

**Review by fmigneault — CHANGES_REQUESTED** [review]

File: src/notebook_utils/automation.py (hunk 1 above)
Comment (antoine-daigneault-demers) [reply src/notebook_utils/automation.py:44]: Not from what I see online... Only functions to check if the path is a file or a directory (`Path.is_file()` or `Path.is_dir()`.

Comment (fmigneault) [reply src/notebook_utils/automation.py:44]: Ok. What I saw was similar to what you did with `ValidateIsDirectory`.

**Review by fmigneault — APPROVED** [review]
All good. Remember to tag after merge and update changelogs.

**Pull request merged**